include/nanobind/stl/filesystem.h

```diff
@@ -0,0 +1,77 @@
+/*
+    nanobind/stl/filesystem.h: type caster for std::string
+
+    Copyright (c) 2023 Qingnan Zhou and Wenzel Jakob
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+#pragma once
+
+#include <nanobind/nanobind.h>
+
+#include <filesystem>
+#include <string>
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+template <>
+struct type_caster<std::filesystem::path> {
+private:
+    static str unicode_from_fs_native(const std::string &w) {
+        PyObject* result = PyUnicode_DecodeFSDefaultAndSize(w.c_str(),
+                                                Py_ssize_t(w.size()));
+        return steal<str>(result);
+    }
+
+    static str unicode_from_fs_native(const std::wstring &w) {
+        PyObject* result = PyUnicode_FromWideChar(w.c_str(), Py_ssize_t(w.size()));
+        return steal<str>(result);
+    }
+
+public:
+    static handle from_cpp(const std::filesystem::path &path, rv_policy,
+                           cleanup_list *) noexcept {
+        if (auto py_str = unicode_from_fs_native(path.native())) {
+            return module_::import_("pathlib")
```

Comment:
The code here is a little naïve -- the `Path()` call might, e.g., raise an exception that would crash the process. Let's write
```cpp
try {
    return module_::import_(..)...
} catch (const python_error &e) {
    e.restore();
}
return handle();
````